app.rb

```diff
@@ -0,0 +1,177 @@
+    new_numbers
+  end
+
+  def my_all?(arg = nil)
+    each do |j|
+      if block_given?
+        return false unless yield j
+      end
+      if !block_given? && arg.nil?
+        return false if j == false || j.nil?
+      end
+      unless arg.nil?
+        if arg.is_a?(Class)
+          return false unless j.is_a?(arg)
+        elsif arg.class == Regexp
+          return false unless arg.match(j)
+        else
+          return false unless arg == j
+        end
+      end
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    each do |j|
+      if block_given?
+        return true if yield j
+      end
+      if arg.nil? && !block_given?
+        return true if j
+      end
+      unless arg.nil?
+        if arg.is_a?(Class)
+          return true if j.is_a?(arg)
+        elsif arg.class == Regexp
+          return true if arg.match(j)
+        else
+          return true unless arg != j
+        end
+      end
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    each do |j|
+      if block_given?
+        return false if yield j
+      end
+      if arg.nil? && !block_given?
+        return false if j
+      end
+      if arg.class == Regexp
+        return false if arg.match(j)
+      end
+      unless arg.nil? && !arg.is_a?(Class)
+        return false if arg == j
+      end
+      if arg.is_a? Class
+        return false if j.is_a?(arg)
+      end
+    end
+    true
+  end
+
+  def my_count(counted = nil)
+    new_array = is_a?(Range) ? to_a : self
+    return new_array.length if counted.nil? && !block_given?
+
+    counting = 0
+    if block_given?
+      new_array.my_each { |j| counting += 1 if yield j }
+    end
+    unless counted.nil?
+      new_array.my_each { |j| counting += 1 if j == counted }
+    end
+    counting
+  end
+
+  def my_map(proc = nil)
+    return enum_for unless block_given?
+
+    arr = []
+    each do |i|
+      if !proc.nil?
+        arr.push(proc.call(i))
+      else
+        arr.push(yield i)
+      end
+    end
+    arr
+  end
+
+  def my_inject(*arg)
+    new_array = is_a?(Range) ? to_a : self
+    product = 0
+    if !block_given?
+      if arg[0].class == Symbol
+        new_array.my_each { |i| product = product.send(arg[0], i) }
+        if product.zero?
+          product = 1
+          new_array.my_each { |i| product = product.send(arg[0], i) }
+        end
+      elsif arg[0].class != Symbol && !arg[1].nil?
+        new_array.my_each { |i| product = product.send(arg[1], i) }
+        if product.zero?
+          product = 1
+          new_array.my_each { |i| product = product.send(arg[1], i) }
+        end
+        product = product.send(arg[1], arg[0])
+      end
+    else
+      new_array.my_each { |i| product = yield product, i }
+      if product.zero?
+        product = 1
+        new_array.my_each { |i| product = yield product, i }
+      end
+      product = yield product, arg[0] unless arg[0].nil?
+    end
+    product
+  end
+end
```

Comment:
# my_inject?
1. [ ] Raises a "LocalJumpError" when no block or argument is given
### Test 👉 ` { array.my_inject }`
**expected `LocalJumpError` but nothing was raised**

2. [ ] Searches for the longest word in an array of strings

```
variable:
  let(:search) { proc { |memo, word| memo.length > word.length ? memo : word } }
```
### Test 👉 `words.my_inject(&search)`
Resulted in:
 _NoMethodError:
       undefined method `length' for 0:Integer_